src/plugins/dnf/product_id.py

```diff
@@ -165,30 +163,45 @@ def _get_available():
         return available
 
     @staticmethod
-    def __write_cache_file(data, file_name):
+    def _write_cache_file(data, file_name):
         try:
             dir_name = os.path.dirname(file_name)
-            if not os.access(dir_name, os.R_OK):
+            cache_dir_exists: bool = False
+            if os.access(dir_name, os.R_OK):
+                cache_dir_exists = True
+            else:
                 log.debug("Try to create directory: %s" % dir_name)
-                os.makedirs(dir_name)
-            with open(file_name, "w") as file:
-                json.dump(data, file, default=json.encode)
-            log.debug("Wrote cache: %s" % file_name)
+                try:
+                    os.makedirs(dir_name, exist_ok=True)
```

Comment:
`exist_ok=True` is not needed here, as the directories definitely do not exist, otherwise the `os.access(dir_name, os.R_OK)` returns True and we do not get to this `os.makedirs()`

Reply:
Yeah, you are right.